client/views/add/add.js

```diff
@@ -9,6 +9,18 @@ function showModsError(reason) {
   currentError = Blaze.renderWithData(Template.form_error, reason, parentNode, nextNode);
 }
 
+function checkPrevMod(modBoxes) {
+  let modEmails = new Set();
+  for (let i = 0; i < modBoxes.length - 1; i++) {
+    modEmails.add(modBoxes[i].value);
+  }
+  const currMail = modBoxes[modBoxes.length - 1].value;
+  if (modEmails.has(currMail)) {
+    showModsError('Email ID was already added as a moderator.');
```

Comment:
Sorry, more nitpicking.

You'll find that functions are easier to reuse if they don't have "side effects." I think of functions as little machines that take a few inputs (the arguments), and spit out a single output (the return value). The little machines have labels on the front (the function name) that tell you exactly what they do. They shouldn't spit out their single output and _also_ do something that's not on the label (the side effect). (These are concepts from [functional programming](https://en.wikipedia.org/wiki/Functional_programming).)

This function should be doing one thing: telling other parts of the code whether or not there are duplicate moderators. If that's what it does, and that's _all_ it does, we can use it all over the place. But if it also tries to show an error message, we can't use it in places where we want to check for duplicates but not necessarily show an error message. Maybe we don't have that need right now, but we might have it in the future. If we write all our code while trying to make sure it'll be generally useful in the future, our future selves will be much faster coders because they can use all the stuff we already built.

That's the long way of saying this function should just return true or false, and the error should be shown down below in the `click` function where we're showing other errors.

Reply:
I totally agree with you on this. thanks